src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/RethrowToPreserveStackDetailsTests.cs

```diff
@@ -92,6 +92,30 @@ End Class
 ");
         }
 
+        [Fact]
+        public async Task CA2200_NoDiagnosticsForThrowAnotherExceptionInWhenClause()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+public abstract class C
+{
+    public void M()
+    {
+        try
+        {
+        }
+        catch (Exception ex) when (Map(ex, out Exception mappedEx))
```

Comment:
Can you add a similar test where we have ‘catch(Exception) when M(out Exception ex)’ followed by ‘throw ex’? That should not fire.